src/tarantool.c

```diff
@@ -631,21 +631,30 @@ int convert_iterator(zval *iter, int all) {
 }
 
 int get_spaceno_by_name(tarantool_connection *obj, zval *name) {
-	if (Z_TYPE_P(name) == IS_LONG)
-		return Z_LVAL_P(name);
-	if (Z_TYPE_P(name) != IS_STRING) {
+	if (Z_TYPE_P(name) != IS_STRING && Z_TYPE_P(name) != IS_LONG) {
```

Comment:
It is strange. We already know space id, wy should we fetch schema in the case?

The problem root is that we check whether a schema is actual only at space nmae resolving, but not in index name resolving. I would just check it in the both places.

Reply:
OTOH, if we'll verify whether a space exist here, we'll give more proper error if it does not exist. Say, not `No index 'name' defined`, but `No space 1024 defined`.

Reply:
> The problem root is that we check whether a schema is actual only at space name resolving, but not in index name resolving. I would just check it in the both places.

I found that it is not accurate, shame on me. We actually fetch an index information in the case, but `schema_add_index()`, which adds the index info assumes that a space information already present.

Reply:
Okay, left the approach as is. Providing an adequate error message is important point.